financial-connections-example/src/main/java/com/stripe/android/financialconnections/example/FinancialConnectionsPlaygroundViewModel.kt

```diff
@@ -278,9 +255,18 @@ class FinancialConnectionsPlaygroundViewModel(
         true,
         emptySet()
     )
+
+    fun onSettingsChanged(playgroundSettings: PlaygroundSettings) {
```

Comment:
So this is updating the entire state with the immutable settings every time it's changed.

I guess I thought about this problem differently. While it's in the UI, keep it mutable (like the UI is mutable), but once it moves into state, take a snapshot of it, make all the mutable functions go away, and have your snapshot in time.

Reply:
I'll definitely iterate over this if I see value over time, the concept of snapshot is definitely valuable! especially as we add more settings in the future.